addon/mixins/authenticated-route-mixin.js

```diff
@@ -80,10 +75,27 @@ export default Mixin.create({
     @default 'login'
     @public
   */
-  authenticationRoute: computed(function() {
-    return Configuration.authenticationRoute;
+  authenticationRoute: computed({
+    get() {
+      if (this.hasOwnProperty('authenticationRouteOverride')) {
+        return this.authenticationRouteOverride;
+      } else {
+        return Configuration.authenticationRoute;
+      }
+    },
+
+    set(key, value) {
+      this.authenticationRouteOverride = value;
+      return value;
+    }
   }),
 
+  init() {
+    this._super(...arguments);
+    let owner = getOwner(this);
+    this._router = owner.lookup('service:router') || owner.lookup('router:main');
```

Comment:
Can't we just inject the router service now ?

Reply:
I don’t think so, given that this addon is compatible back to Ember 1.12.

Reply:
We should release a 2.0 soon to get rid of all this ancient stuff 😞

Reply:
@backspace the problem with this line is that it will always call `owner.lookup`, even when the test overrides the `_router` property (which is a problem in itself but shouldn't be refactored as part of this PR), e.g. https://github.com/simplabs/ember-simple-auth/blob/master/tests/unit/mixins/authenticated-route-mixin-test.js#L64

If you change this to

```js
this._router = this._router || owner.lookup('service:router') || owner.lookup('router:main');
```

it works as expected.